shared/src/main/java/uk/gov/di/authentication/shared/helpers/CookieHelper.java

```diff
@@ -166,7 +175,7 @@ public static String appendTimestampToCookieValue(String cookieValue) {
     }
 
     public static boolean isValidCookieWithDoubleDashedTimestamp(String cookieValue) {
-        String sessionIdPattern = ".*--\\d+";
+        String sessionIdPattern = ".+--\\d+[^-]*";
```

Comment:
Would it be correct to have `^.+--\\d+$` ? I'm struggling with the intent of the `[^-]*` above a little

Reply:
It's to check that we haven't generated a double timestamp like some of what we've seen 'in the wild' e.g. `--1700558480962--1700558480963` as a persistent-session-id. That's the intent. As to whether it's necessary, up to you?

Reply:
I've gone with the same regex as elsewhere where you'd recommended that - this broke some tests since we commonly used something like `a-persistent-session-id` as a test value, which doesn't match the format requirement, but no bad thing to make our tests use a 'proper' ID?

Reply:
This regex would still match a double timestamp. The leading `.+` would match the first timestamp, and the `--\\d+` would match the second.

Reply:
Ah yes 🤦